cts/lab/CTStests.py

```diff
@@ -1299,9 +1303,31 @@ def choose_resource(self, node, resourcelist):
                     return rsc
         return None
 
+    def get_failcount(self, node):
+        """ Check the fail count of targeted resource on given node """
+
+        (rc, lines) = self.rsh(node,
+                               "crm_failcount --quiet --query --resource %s "
+                               "--operation %s --interval %d "
+                               "--node %s" % (self.rid, self.action,
+                               self.interval, node), None)
+        if rc != 0 or len(lines) != 1:
+            self.logger.log("crm_failcount on %s failed (%d): %s" % (node, rc,
+                            ' '.join(lines)))
```

Comment:
Does each line have a trailing newline or not? If so, then this should give us a bunch of lines, all but the first of which starts with a space character. Alternative would be to strip the lines and join on the empty string.

Or if the lines don't have trailing newlines, we could join on `'\n'` instead of `' '` -- depends on the log message format you want.

Reply:
Sounds good to me. Some tools log literal `'\n'` as a string in the log messages instead of expanding it to a newline.

> Alternative would be to strip the lines and join on the empty string.

Re-reading this, I must have meant join on `'\n'`, since `str.strip` removes newlines as well as spaces.